### PR TITLE
Login and roster upload

### PR DESCRIPTION
**Pull request by jrm86**
Implementation of the Login and Roster Upload functionality.

Includes the authentication stub written by @afig to facilitate login.

### CODE REVIEW

**Comment by DasilvaB** [comment]
In index.js on line 138 you have:

`reader.onload = uploadRoster;`

I'm assuming you meant to call the method, so would need the () and to pass in the event? Also inside the method uploadRoster on line 145 its missing a space.

File: src/webapp/client/js/index.js [hunk 1]
```diff
@@ -121,8 +129,25 @@ $(document).ready(function() {
 		$('#rosterTab, #attnTab, #gradesTab, #reportsTab').css('display', 'none');
 		$('ul.tabs').tabs('select_tab', 'login');
 	});
+
+	$('#uploadBtn').click(function()
+	{
+		var file = document.getElementById('#rosterImport').files[0];
```
Comment (afig) [inline src/webapp/client/js/index.js]:
```suggestion
		dbInfo = getDBFields();
		var file = document.getElementById('rosterImport').files[0];
```

Comment (afig) [reply src/webapp/client/js/index.js]: The `#` is not necessary when using `document.getElementByID()` (only when using `$()`)

Also, the database info need to also be send with the uploaded file, so this is part 1 of transmitting that info.

File: src/webapp/client/js/index.js [hunk 2]
```diff
@@ -121,8 +129,25 @@ $(document).ready(function() {
...
+		var reader = new FileReader();
+		reader.readAsText(file, 'UTF-8');
+		reader.onload = uploadRoster;
+	});
+
 });
 
+function uploadRoster(event){
+	var result = event.target.result;
+    var fileName = document.getElementById('#rosterImport').files[0].name;
```
Comment (afig) [inline src/webapp/client/js/index.js]:
```suggestion
	var fileName = document.getElementById('rosterImport').files[0].name;
```

Comment (afig) [reply src/webapp/client/js/index.js]: Changed spaces to tab and removed unnecessary `#`.

File: src/webapp/client/js/index.js [hunk 3]
```diff
@@ -121,8 +129,25 @@ $(document).ready(function() {
...
+	
+	$.post('/importSectionRoster', { data: result, name: fileName }, showAlert("<p>Upload Successful.</p>"));
```
Comment (afig) [inline src/webapp/client/js/index.js]:
```suggestion
	var data = $.extend({}, dbInfo, {data: result}, {name: fileName});
	$.post('/importSectionRoster', data, showAlert("<p>Upload Successful.</p>"));
```

Comment (afig) [reply src/webapp/client/js/index.js]: As mentioned in an earlier suggestion, the DB info needs to be included in the upload. Here, a new object is being created that includes both the data being uploaded as well as the connection info.

**Review by afig — COMMENTED** [review]
Added some suggestions that address issues with file uploads on the client side. There are a few more things that needed to be addressed on the middleware side to allow uploads to work correctly. I can discuss those tomorrow.

**Comment by jrm86** [comment]
Thank you for the suggestions and fixes @afig.  Had some issues with my local install of the DB portion, which made testing a little troublesome.  We'll discuss this and other changes during today's meeting.

**Review by afig — APPROVED** [review]
Looks good to merge. I added a commit that addressed the merge conflict.

**Review by DasilvaB — APPROVED** [review]
Nice job, looks good.

**Pull request merged**